### ISSUE TITLE
enums don't default to UNRECOGNIZED value like java

### ISSUE THREAD

**Issue by mscolnick** [stephenh/ts-proto#41]
To be compatible with newly added enums in proto definitions, the typescript enum deserialization should default to `UNRECOGNIZED` / `-1`

https://github.com/stephenh/ts-proto/blob/master/src/main.ts#L321

This is what java generates for enum

```java
public enum ChangeType implements ProtocolMessageEnum {
    ADDED(0),
    REMOVED(1),
    MODIFIED(2),
    UNRECOGNIZED(-1);
```

Happy to make a contribution if you are ok with the update.

**Comment by stephenh** [comment]
Ah yeah, good point, that makes sense. A PR would be great.

Just a heads up, I very recently moved away from `enum` the keyword to this `enum`-via-`const`+`type` pattern b/c babel didn't like using `namespace`s to add static methods, i.e. `FooEnum.toJSON(foo)`, but otherwise, the output/API still looks very enum-ish, so should be pretty amenable to your feature request/PR.

**Comment by mscolnick** [comment]
makes total sense. will put something up within the week. thank you

**Comment by stephenh** [comment]
I merged and pushed to npm release 1.15.0 with this change. Thanks for the PR!
